app/models/acts_as_constrained/concerns/constrained.rb

```diff
@@ -56,12 +62,35 @@ def self.constrain_by constraint_type, options = {}
       #
       # MyConstrainedModel.constrained_by_date(Date.tomorrow).
       #   constrained_by_dummy(99)
+      #
+      # Some constrains may be optional. In that case it will try to get results with them (or fallback to the
+      # the required ones). The concatenation of more scopes will not work in this scenario.
```

Comment:
In fact, this scope returns an ActiveRecord Relation if the constrained_by scopes defined by the users of the gem return an ActiveRecord Relation. Doesn't it?